File: Flutter/App/lib/presentation/product/model/presentation_product.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'presentation_product.freezed.dart';

@freezed
class PresentationProduct with _$PresentationProduct {
  const factory PresentationProduct({
    required int id,
    required String title,
    required String description,
    required double price,
    required String category,
    required List<String> images,
  }) = _PresentationProduct;
}

File: iOS/App/ProductsCrud/Presentation/Product/Model/PresentationProduct.swift
//
//  PresentationProduct.swift
//  ProductsCrud
//
//  Created by Emil Falkowski on 11/11/2024.
//

struct PresentationProduct: Identifiable {
    let id: Int
    let title: String
    let description: String
    let price: Double
    let category: String
    let images: [String]
}
